base/applications/taskmgr/graphctl.h

```diff
@@ -22,93 +22,61 @@
 
 #pragma once
 
-#define MAX_PLOTS 4
-#define MAX_CTRLS 4
+#define NUM_PLOTS    2
+#define PLOT_SHIFT   2
 
 #ifdef __cplusplus
 extern "C" {
 #endif
 
-#if 0
-
-/* Attributes */
-public:
-  void SetXUnits(const char* string);
-  void SetYUnits(const char* string);
-
-  /* Operations */
-public:
-  BOOL Create(DWORD dwStyle, const RECT& rect, HWND hParentWnd, UINT nID=NULL);
+extern WNDPROC OldGraphCtrlWndProc;
 
-#endif
+INT_PTR CALLBACK GraphCtrl_WndProc(HWND hWnd, UINT message, WPARAM wParam, LPARAM lParam);
 
-typedef struct
+typedef struct _TM_GRAPH_CONTROL
 {
-  int m_nShiftPixels;          /* amount to shift with each new point */
-  int m_nYDecimals;
-
-  char m_strXUnitsString[50];
-  char m_strYUnitsString[50];
-
-  COLORREF m_crBackColor;                 /* background color */
-  COLORREF m_crGridColor;                 /* grid color */
-  COLORREF m_crPlotColor[MAX_PLOTS];      /* data color   */
-
-  double m_dCurrentPosition[MAX_PLOTS];   /* current position */
-  double m_dPreviousPosition[MAX_PLOTS];  /* previous position */
-
-/* those were protected fields */
-  int m_nHalfShiftPixels;
-  int m_nPlotShiftPixels;
-  int m_nGridOffsetPixels;
-  int m_nClientHeight;
-  int m_nClientWidth;
-  int m_nPlotHeight;
-  int m_nPlotWidth;
-  int m_nGridXPixels;
-  int m_nGridYPixels;
-
-  double m_dLowerLimit;        /* lower bounds */
-  double m_dUpperLimit;        /* upper bounds */
-  double m_dRange;
-  double m_dVerticalFactor;
-
-  HWND     m_hWnd;
-  HWND     m_hParentWnd;
-  HDC      m_dcGrid;
-  HDC      m_dcPlot;
-  HBITMAP  m_bitmapOldGrid;
-  HBITMAP  m_bitmapOldPlot;
-  HBITMAP  m_bitmapGrid;
-  HBITMAP  m_bitmapPlot;
-  HBRUSH   m_brushBack;
-  HPEN     m_penPlot[MAX_PLOTS];
-  RECT     m_rectClient;
-  RECT     m_rectPlot;
-} TGraphCtrl;
+    HWND     hParentWnd;
+    HWND     hWnd;
+    HDC      hdcGraph;
+    HBITMAP  hbmGraph;
+    HPEN     hPenGrid;
+    HPEN     hPen0;
+    HPEN     hPen1;
+    HBRUSH   hBrushBack;
+
+    INT      BitmapWidth;
+    INT      BitmapHeight;
+    INT      GridCellWidth;
+    INT      GridCellHeight;
+    INT      CurrShift;
+    INT      Pad;
+
+    PBYTE    PointBuffer;
```

Comment:
Could be declared as
```suggestion
    PBYTE    PointBuffer[NUM_PLOTS];
```
so that each plot has its point buffer as `PointBuffer[indexPlot];`.

Reply:
argh... no, the way it's done, use just 1 allocation/deallocation. less function calls.